### PR TITLE
implement title's value from github rest api

### PR DESCRIPTION
**Pull request by FrostEvil**

### CODE REVIEW

File: src/components/CardProject.astro [hunk 1]
```diff
@@ -1,54 +1,58 @@
...
-        )
+        // skills.map((skill) =>
+        //   mapSkillToCss[skill] ? (
+        //     <li class={mapSkillToCss[skill]}>{skill}</li>
+        //   ) : (
+        //     <li class="text-cyan-600">{skill}</li>
+        //   )
+        // )
       }
     </ul>
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:57]: Any idea on getting the skills from repo data?

File: src/pages/index.astro [hunk 2]
```diff
@@ -3,60 +3,79 @@ import Layout from "../layouts/Layout.astro";
...
+// const titles = repos.data.map((data) => Object.values(data["name"]).join(""));
 
-  title: "Real Estate",
-  description:
-    "Introducing Real Estate: Your trusted partner in real estate management. From residential complexes to commercial spaces, we offer comprehensive property management services tailored to meet your needs. With a focus on efficiency, transparency, and maximizing property value.",
-  skills: baseSkills,
-};
+const titles = repos.data.map((data) => {
+  return data["name"];
+});
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:25]: Does it look similar to the map we do in the template?

File: src/pages/index.astro [hunk 3]
```diff
@@ -3,60 +3,79 @@ import Layout from "../layouts/Layout.astro";
...
+// const forestCard = {
+//   img: {
+//     src: `${import.meta.env.BASE_URL}/project_4.png`,
+//     alt: "Forest Group welcome page",
+//   },
+//   title: "Forest Group",
+//   description:
+//     "Welcome to Forest Group: Your digital oasis for landscaped parks. Explore lush green spaces, tranquil gardens, and vibrant landscapes from the comfort of your screen. Discover inspiration for your next outdoor adventure, family picnic, or leisurely stroll.",
+//   skills: baseSkills,
+// };
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:70]: We can remove it

File: src/pages/index.astro [hunk 4]
```diff
@@ -3,60 +3,79 @@ import Layout from "../layouts/Layout.astro";
...
 
 const modalContent = {
   title: "Checking title",
   description:
     "Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio praesentium quis veritatis repellendus quam. Quam atque ad sapiente similique accusantium pariatur veniam ipsum reprehenderit, aliquam tenetur recusandae excepturi, aut architecto?",
 };
 
-const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
+// const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:78]: Same here

File: src/pages/index.astro [hunk 5]
```diff
@@ -123,3 +142,20 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
...
+      per_page: 5,
+    })
+    .then((result) => {
+      return { owner: "bright-bounty", repo: result };
+    })
+    .catch((error) =>
+      console.error("There was a problem with octokit request", error)
+    );
+}
+getGithubUser(); -->
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:161]: Not needed anymore

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Looks better, now we have same upgrades to do in order to move on

File: src/pages/index.astro [hunk 6]
```diff
@@ -3,60 +3,79 @@ import Layout from "../layouts/Layout.astro";
...
-    alt: "Weather App Layout",
-  },
-  title: "Weather App",
-  description:
-    "Stay ahead of the forecast with our intuitive weather app. Get real-time updates on current conditions, hourly forecasts, and daily outlooks. Plan your day confidently with accurate weather information at your fingertips. Never be caught off guard by the elements again!",
-  skills: baseSkills,
-};
+const octokit = new Octokit({
+  auth: import.meta.env.OCTOKIT_TOKEN,
+  userAgent: "My Portfolio",
```
[the remark this answers is not in the document]
Comment (FrostEvil) [reply src/pages/index.astro:11]: That is subpoint 3 from sav-20 task. Maybe userAgent is not required.

File: src/components/CardProject.astro [hunk 7]
```diff
@@ -1,54 +1,58 @@
...
-  CSS: "text-blue-600",
-  JavaScript: "text-yellow-600",
-};
+const { title } = Astro.props;
 ---
 
 <div class="p-6 flex flex-col items-center gap-x-6 shadow-md">
   <div class="h-64">
-    <img class="rounded-3xl w-64 h-64" src={src} alt={alt} />
+    <img class="rounded-3xl w-64 h-64" />
```
[the remark this answers is not in the document]
Comment (FrostEvil) [reply src/components/CardProject.astro:38]: You mean that I need to use one of my image here?

File: src/pages/index.astro (hunk 3 above)
Comment (FrostEvil) [reply src/pages/index.astro:70]: Removed

File: src/pages/index.astro (hunk 4 above)
Comment (FrostEvil) [reply src/pages/index.astro:78]: Removed

File: src/pages/index.astro (hunk 5 above)
Comment (FrostEvil) [reply src/pages/index.astro:161]: Removed

File: src/pages/index.astro (hunk 2 above)
Comment (FrostEvil) [reply src/pages/index.astro:25]: Yes, I changed map in the template and used repos.data there.

File: src/components/CardProject.astro [hunk 8]
```diff
@@ -1,54 +1,58 @@
...
   </div>
   <div class="relative h-full pb-8">
     <h3
       class="mt-6 text-lg font-semibold leading-7 tracking-tight text-gray-900"
     >
       {title}
     </h3>
-    <p class="text-base font-semibold leading-6 text-slate-700 my-8">
-      {description}
-    </p>
```
[the remark this answers is not in the document]
Comment (FrostEvil) [reply src/components/CardProject.astro:42]: Done

File: src/components/CardProject.astro (hunk 1 above)
Comment (FrostEvil) [reply src/components/CardProject.astro:57]: I did it kind a title and description but additionally with adding color to skill value.
But it is any possible to get more than one language here from GH Rest Api? From repos.data["language"] I suppose to get only one.

File: src/components/CardProject.astro [hunk 9]
```diff
@@ -29,7 +16,11 @@ const mapSkillToCss = {
 
 <div class="p-6 flex flex-col items-center gap-x-6 shadow-md">
   <div class="h-64">
-    <img class="rounded-3xl w-64 h-64" src={src} alt={alt} />
+    <img
+      class="rounded-3xl w-64 h-64"
+      src=`${import.meta.env.BASE_URL}/project_4.png`
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:21]: via.placeholder.com

File: src/pages/index.astro [hunk 10]
```diff
@@ -110,9 +74,13 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
         <div class="mt-24">
           <ul class="grid md:grid-cols-2 gap-8 md:gap-16">
             {
-              cards.map((card) => (
+              repos.data.map((data) => (
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:77]:
```suggestion
              repos.data.map((repo) => (
```

File: src/pages/index.astro [hunk 11]
```diff
@@ -110,9 +74,13 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
...
                 <li class="flex justify-stretch">
-                  <CardProject card={card} />
+                  <CardProject
+                    title={data["name"]}
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:80]:
```suggestions 
                    title={data.name"]}
```

File: src/pages/index.astro [hunk 12]
```diff
@@ -110,9 +74,13 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
...
+                    description={data["description"]}
+                    skill={data["language"]}
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:82]:
```suggestion
                    description={data.description}
                    skill={data.language}
```

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]

File: src/pages/index.astro (hunk 10 above)
Comment (FrostEvil) [reply src/pages/index.astro:77]: changed

File: src/pages/index.astro (hunk 11 above)
Comment (FrostEvil) [reply src/pages/index.astro:80]: changed

File: src/pages/index.astro (hunk 12 above)
Comment (FrostEvil) [reply src/pages/index.astro:82]: changed

File: src/components/CardProject.astro (hunk 9 above)
Comment (FrostEvil) [reply src/components/CardProject.astro]: added

File: src/pages/index.astro (hunk 6 above)
[the remark this answers is not in the document]
Comment (aleksanderwalczuk) [reply src/pages/index.astro]: `My Portfolio` is not a user agent

File: src/components/CardProject.astro (hunk 1 above)
Comment (aleksanderwalczuk) [reply src/components/CardProject.astro]: Already solved

File: src/components/CardProject.astro [hunk 13]
```diff
@@ -1,35 +1,40 @@
...
+      src="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2264%22%20height%3D%2264%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2064%2064%22%20preserveAspectRatio%3D%22none%22%3E%0A%20%20%20%20%20%20%3Cdefs%3E%0A%20%20%20%20%20%20%20%20%3Cstyle%20type%3D%22text%2Fcss%22%3E%0A%20%20%20%20%20%20%20%20%20%20%23holder%20text%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20fill%3A%20%23000000%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20font-family%3A%20sans-serif%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20font-size%3A%206px%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20font-weight%3A%20500%3B%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%3C%2Fstyle%3E%0A%20%20%20%20%20%20%3C%2Fdefs%3E%0A%20%20%20%20%20%20%3Cg%20id%3D%22holder%22%3E%0A%20%20%20%20%20%20%20%20%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22%231cd5ff%22%3E%3C%2Frect%3E%0A%20%20%20%20%20%20%20%20%3Cg%3E%0A%20%20%20%20%20%20%20%20%20%20%3Ctext%20text-anchor%3D%22middle%22%20x%3D%2250%25%22%20y%3D%2250%25%22%20dy%3D%22.3em%22%3EPlaceholder%20img%3C%2Ftext%3E%0A%20%20%20%20%20%20%20%20%3C%2Fg%3E%0A%20%20%20%20%20%20%3C%2Fg%3E%0A%20%20%20%20%3C%2Fsvg%3E"
```
Comment (aleksanderwalczuk) [inline src/components/CardProject.astro:35]: Why base64?

**Review by aleksanderwalczuk — APPROVED** [review]

**Pull request merged**